XMonad/Hooks/StatusBar.hs

```diff
@@ -0,0 +1,422 @@
+{-# LANGUAGE FlexibleContexts, TypeApplications #-}
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  XMonad.Hooks.StatusBar
+-- Copyright   :  (c) Yecine Megdiche <yecine.megdiche@gmail.com>
+-- License     :  BSD3-style (see LICENSE)
+--
+-- Maintainer  :  Yecine Megdiche <yecine.megdiche@gmail.com>
+-- Stability   :  unstable
+-- Portability :  unportable
+--
+-- This module provides a new interface that replaces "XMonad.Hooks.DynamicLog",
+-- by providing composoble and dynamic status bars. Supports property-based as well
+-- as pipe-based status bars.
+--
+-----------------------------------------------------------------------------
+
+module XMonad.Hooks.StatusBar (
+  -- * Usage
+  -- $usage
+  StatusBarConfig(..),
+  makeStatusBar,
+  makeStatusBar',
+
+  -- * Available Configs
+  -- $availableconfigs
+  statusBarPipe,
+  statusBarProp,
+  statusBarPropTo,
+
+  -- * Multiple Status Bars
+  -- $multiple
+
+  -- * Property Logging utilities
+  xmonadPropLog,
+  xmonadPropLog',
+  xmonadDefProp,
+
+  -- * Managing status bar Processes
+  -- $sbprocess
+  spawnStatusBarAndRemember,
+  cleanupStatusBars,
+  ) where
+
+import Control.Exception (SomeException, try)
+import Control.Monad (void)
+import qualified Codec.Binary.UTF8.String as UTF8 (encode)
+import System.Posix.Signals (sigTERM, signalProcessGroup)
+import System.Posix.Types (ProcessID)
+
+import qualified Data.Map        as M
+
+import Foreign.C (CChar)
+
+import XMonad
+
+import XMonad.Util.Run
+import qualified XMonad.Util.ExtensibleState as XS
+
+import XMonad.Layout.LayoutModifier
+import XMonad.Hooks.ManageDocks
+import XMonad.Hooks.StatusBar.PP
+
+-- $usage
+-- You can use this module with the following in your @~\/.xmonad\/xmonad.hs@:
+--
+-- >    import XMonad
+-- >    import XMonad.Hooks.StatusBar
+-- >    import XMonad.Hooks.StatusBar.PP
+--
+-- The recommended way to use this module with xmobar, as well as any other
+-- status bar that supports property logging (you can read more about X11
+-- properties
+-- [here](https://en.wikipedia.org/wiki/X_Window_System_core_protocol#Properties)
+-- or
+-- [here](https://tronche.com/gui/x/xlib/window-information/properties-and-atoms.html),
+-- although you don't have to understand them in order to use the functions
+-- below), is to use 'statusBarProp' with 'makeStatusBar', which takes care of
+-- the necessary plumbing-no shell scripting required!
+--
+-- > main = do
+-- >   mySB <- statusBarProp "xmobar" (pure myPP)
+-- >   xmonad =<< makeStatusBar mySB myConf
+--
+-- which plays nice with other combinators that you might have already
+-- in your config:
+--
+-- > main = do
+-- >   mySB <- statusBarProp "xmobar" (pure myPP)
+-- >   xmonad =<< (makeStatusBar mySB . ewmh . docks $ def {...})
+--
+-- It is generally a good idea to also add
```

Comment:
Something seems to be missing here. :-)

(Please, from this point on, push fixup commits instead of forcepushing. I'm hacking on it too. :-))

Reply:
I have absolutely no idea what I wanted to say